rootfs/deploy.py

```diff
@@ -36,6 +36,39 @@ def log(msg):
         print(msg)
 
 
+def get_registry_name():
+    hostname = os.getenv('DEIS_REGISTRY_HOSTNAME', "").replace("https://", "")
+    if regsitryLocation == "off-cluster":
+        organization = os.getenv('DEIS_REGISTRY_ORGANIZATION')
+        regName = ""
+        if hostname != "":
+            regName = hostname + "/"
+        if organization != "":
+            regName = regName + organization
+        return regName
+    elif regsitryLocation == "ecr":
+        return hostname
+    elif regsitryLocation == "gcr":
+        return hostname+"/"+os.getenv('DEIS_REGISTRY_GCS_PROJ_ID')
+    else:
+        return os.getenv("DEIS_REGISTRY_SERVICE_HOST") + ":" + os.getenv("DEIS_REGISTRY_SERVICE_PORT")
+
+
+def docker_push(client, repo, tag):
+    if regsitryLocation != "on-cluster":
+        hostname = os.getenv('DEIS_REGISTRY_HOSTNAME')
+        if hostname == '':
+            hostname = "https://index.docker.io/v1/"
```

Comment:
if the registry location is on-cluster, why would we default to dockerhub? Shouldn't we error out here instead?